pkg/resources/resources_manager_test.go

```diff
@@ -0,0 +1,290 @@
+				ResourceName: "test_config$$",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: resource name \"test_config$$\" contains invalid characters"))
+		})
+		It("resources with repeated names", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   500,
+				Devices:      []string{"ib1"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: resource name \"test_config\" already exists"))
+		})
+		It("resources with invalid number of resources", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   -100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: Invalid value for rdmaHcaMax < 0: -100"))
+		})
+		It("resources with no devices", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: no devices provided"))
+		})
+	})
+	Context("InitServers", func() {
+		It("Init valid server", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.InitServers()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(rm.configList)).To(Equal(1))
+		})
+		It("Init server with invalid number of resources", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   -100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.InitServers()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("StartAllServers", func() {
+		It("start valid server", func() {
+			fakeDevicePlugin := mocks.ResourceServer{}
+			fakeDevicePlugin.On("Start").Return(nil)
+			fakeDevicePlugin.On("Watch").Return(nil)
+
+			rm := &resourceManager{resourceServers: []types.ResourceServer{&fakeDevicePlugin}}
+
+			err := rm.StartAllServers()
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
you also expect Start() to be called for fakeDevicePlugin
and Watch() to be called if rm.watchMode == True

So this should be split to two testcases

Reply:
Updated with 2 cases when: watchMode = true and watchMode = false